telemetry/statsd/src/main/scala/io/buoyant/telemetry/StatsDInitializer.scala

```diff
@@ -0,0 +1,56 @@
+package io.buoyant.telemetry
+
+import com.timgroup.statsd.NonBlockingStatsDClient
+import com.twitter.finagle.Stack
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.logging.Logger
+import io.buoyant.telemetry.statsd.{StatsDStatsReceiver, StatsDTelemeter}
+import java.util.UUID
+
+class StatsDInitializer extends TelemeterInitializer {
+  type Config = StatsDConfig
+  val configClass = classOf[StatsDConfig]
+  override val configId = "io.l5d.statsd"
+}
+
+private[telemetry] object StatsDConfig {
+  val DefaultPrefix = "linkerd"
+  val DefaultHostname = "127.0.0.1"
+  val DefaultPort = 8125
+  val DefaultGaugePeriodMs = 10000
+
+  val MaxQueueSize = 10000
+  val ProcessId = "l5d-uuid-" + UUID.randomUUID().toString
+}
+
+case class StatsDConfig(
+  prefix: Option[String],
+  hostname: Option[String],
+  port: Option[Int],
+  gaugePeriodMs: Option[Int]
+) extends TelemeterConfig {
+  import StatsDConfig._
+
+  private[this] val log = Logger.get("io.l5d.statsd")
+
+  // prefix format is: "prefix_l5d-uuid-process-id"
+  val statsDPrefix = prefix.getOrElse(DefaultPrefix) + "_" + ProcessId
```

Comment:
I agree with @leozc, this shouldn't be included in metric name by default, as it makes initial kick-off more complex: one should explicitly configure statsd/telegraf to skip prefix to be able to aggregate stats from multiple instances, which is (at least to us) the main goal of using statsd for linkerd monitoring. To distinguish between instances it is better to use tags. Another thing to consider: it is [not recommended](https://docs.influxdata.com/influxdb/v1.1/concepts/schema_and_data_layout/#don-t-have-too-many-series) to have tags with unlimited number of possible values or encode this data into measurement name, known as hitting series cardinality limit. This will eventually put InfluxDB instance into extremely slow and memory consuming state. In other words, instance id should be if not disabled at all, then at least not included in measurement name/tags by default.